client/webserver/site/src/js/markets.ts

```diff
@@ -1770,51 +1770,98 @@ export default class MarketsPage extends BasePage {
     const { page, market } = this
     let { baseUnitInfo, quoteUnitInfo, rateConversionFactor } = market
     const swapped = swap.estimate.value || 0
-    const fmtPct = percentFormatter.format
+    const fmtPct = (value: number) => {
+      if (value < 0.05) return '< 0.1'
+      return percentFormatter.format(value)
+    }
+
+    // If the asset is a token, in order to calculate the fee as a percentage
+    // of the total order, we try to  use the fiat rates to find out the
+    // exchange rate between the token and parent assets.
+    // Initially these are set to 1, which we would use if the asset is not a
+    // token and no conversion is needed.
+    let baseExchangeRate = 1
+    let quoteExchangeRate = 1
 
     if (market.base.token) {
       const parent = app().assets[market.base.token.parentID]
       baseUnitInfo = parent.unitInfo
+
+      const tokenFiatRate = app().fiatRatesMap[market.base.id]
+      const parentFiatRate = app().fiatRatesMap[parent.id]
+      if (tokenFiatRate && parentFiatRate) {
+        baseExchangeRate = tokenFiatRate / parentFiatRate
```

Comment:
I think you need a `* quoteUnitInfo.Conventional.ConversionFactor / baseUnitInfo.Conventional.ConversionFactor` here. Fiat rates use conventional units, but we're operating on atoms below. 

Say, for instance, we're trading on ETH-USDC, and the going rate is 1 ETH for 1 USDC, and that's the exact order we place. Fees are 1e4 gwei, miniscule in comparison to trade size. In that case, `baseExchangeRate` here will be 1, and applying that below to `swapped` would do nothing, resulting in 
```Go
pct = 1e4 /* fees */ / (1e4 /* swapped atoms */ * 1 /* xc rate */ ) * 100 = 100%
```
which is clearly wrong.